### PR TITLE
Add hooks to overwrite atmos albedo 

### LINKED ISSUE #653: overwrite atmos albedo
As of our update to ClimaAtmos v0.21.0 ([PR](https://github.com/CliMA/ClimaCoupler.jl/pull/639)), ClimaAtmos is using `ConstantAlbedo`, and it doesn't get updated in coupled runs. We want to overwrite this albedo in the coupler with the albedo provided by the land model

Diffuse and direct albedo will also be needed bu the Big leaf model, so we implement a more general solution here. 

- [x] atmos hook for the coupler - https://github.com/CliMA/ClimaAtmos.jl/issues/2746
- [x] implement coupler hook
- [x] test expected behaviour
  - [x] slabplanet no change
  - [x] AMIP ocean albedo function of latitude, show that land and ice albedo is not overwritten
  
Screen Shot 2024-03-19 at 9 36 38 AM

- [x] modify docs

Components of the implementation
- 2 new coupler fields for `albedo_direct` and `albedo_diffuse`, as well as in the stub models
- generalize the get_field functions to these two types of albedo 
- 2 new temp fields for dynamic calculations, avoiding allocation on the fly (this can be used in other places in the code to improve performance) see #584 
- new function for `water_albedo_from_wind!` in the FluxCalculator and an extended method for ClimaAtmos

### PR DESCRIPTION
**Pull request by LenkaNovak**
Closes #653 

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: experiments/AMIP/components/atmosphere/climaatmos.jl [hunk 1]
```diff
@@ -384,3 +391,38 @@ function dss_state!(sim::ClimaAtmosSimulation)
...
     end
 end
+
+# albedo extension # the atmos albedo should be moved to the ClimaUtilities?
+"""
+    water_albedo_from_atmosphere!(atmos_sim::ClimaAtmosSimulation, direct_albedo::ClimaCore.Fields.Field, diffuse_albedo::ClimaCore.Fields.Field)
+
+Extension to calculate the water surface albedo from wind speed and insolation. It can be used for prescribed ocean and lakes.
+"""
+function water_albedo_from_atmosphere!(
```
Comment (szy21) [inline experiments/AMIP/components/atmosphere/climaatmos.jl:401]: I'm guessing this means we can't use the `set_surface_albedo!` in atmos. Do you know why?

File: test/field_exchanger_tests.jl [hunk 2]
```diff
@@ -181,18 +196,21 @@ for FT in (Float32, Float64)
...
                 ρ_sfc = Fields.ones(boundary_space),
+                albedo_direct = Fields.ones(boundary_space),
+                albedo_diffuse = Fields.ones(boundary_space) .* FT(0.5),
             )),
         )
 
         # test the sim update under CombinedStateFluxes (update all) and PartitionedStateFluxes (update all except turbulent fluxes)
         flux_types = (CombinedStateFluxes(), PartitionedStateFluxes())
-        results = [FT(0), FT(1)]
+        results = [FT(0), FT(1), FT(0.5)]
```
Comment (szy21) [inline test/field_exchanger_tests.jl:206]: It seems this newly added results[3] is not used in the following tests?

**Review by szy21 — APPROVED** [review]
Thanks! I just have a few minor questions

File: experiments/AMIP/Manifest.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # This file is machine-generated - editing it directly is not advised
 
-julia_version = "1.10.2"
```
[the remark this answers is not in the document]
Comment (LenkaNovak) [reply experiments/AMIP/Manifest.toml:3]: Nope, it's just my local version! Good catch! :)

File: test/field_exchanger_tests.jl (hunk 2 above)
Comment (LenkaNovak) [reply test/field_exchanger_tests.jl:206]: Thanks for flagging this -  I'm still adding a couple of tests after the codecov failed. :)

**Pull request merged**